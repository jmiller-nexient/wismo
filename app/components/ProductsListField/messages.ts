import { defineMessages } from 'react-intl';

export const scope = 'wismo.components.products-list-field';

export default defineMessages({
  productsListFieldLabel: {
    id: `${scope}.productsListFieldLabel`,
    defaultMessage: 'Products',
  },
});
