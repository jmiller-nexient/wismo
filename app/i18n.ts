/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
import { DEFAULT_LOCALE } from './locales';

const enTranslationMessages = require('translations/en.json');
const esTranslationMessages = require('translations/es.json');

// The purpose of this method is to apply fallback logic when a translation
// is not defined for the current locale.
// In our case, we do not want any fallback logic.
// tslint:disable-next-line: typedef
const formatTranslationMessages = (locale: string, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages) : {};

  // tslint:disable-next-line: typedef
  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE ? defaultFormattedMessages[key] : messages[key];

    return { ...formattedMessages, [key]: formattedMessage };
  };

  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages),
  es: formatTranslationMessages('es', esTranslationMessages),
};
