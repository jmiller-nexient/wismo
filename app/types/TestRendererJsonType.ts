import { ReactTestRendererJSON } from 'react-test-renderer';

type TestRendererJsonType = ReactTestRendererJSON | null;

export default TestRendererJsonType;
