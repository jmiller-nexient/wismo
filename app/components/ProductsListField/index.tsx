import * as React from 'react';

import BaseFormField from 'components/BaseField';
import ProductList from 'components/ProductList';

import IProduct from 'types/product';

import messages from './messages';

interface IProductsListFieldProps {
  products: (IProduct[] | null);
}

const ProductsListField: React.FC<IProductsListFieldProps> = (props: IProductsListFieldProps): React.ReactElement => {
  const { products } = props;

  return (
    <BaseFormField label={messages.productsListFieldLabel}>
      <ProductList products={products}/>
    </BaseFormField>
  );
};

export default ProductsListField;
