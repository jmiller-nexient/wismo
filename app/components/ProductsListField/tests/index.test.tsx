import React from 'react';
import renderer from 'react-test-renderer';

import { IntlProvider } from 'react-intl';

import { translationMessages } from '../../../i18n';

import ProductsListField from 'components/ProductsListField';

import IProduct from 'types/product';
import TestRendererJsonType from 'types/TestRendererJsonType';

const fakeProducts: IProduct[] = [
  {
    productType: 'IAC',
    serviceDate: '04-22-2020'
  },
  {
    productType: 'RES_GAS',
    serviceDate: '04-20-2020'
  }
];

describe('<ProductsListField />', () => {
  it('should render correctly when a valid product list is passed', () => {
    const tree: TestRendererJsonType = renderer.create(
      <IntlProvider locale="en" messages={translationMessages.en}>
        <ProductsListField products={fakeProducts}/>
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
