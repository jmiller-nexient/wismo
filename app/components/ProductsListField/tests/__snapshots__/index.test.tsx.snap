// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsListField /> should render correctly when a valid product list is passed 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    Products
  </label>
  <div>
    <span
      className="dte-wismo-base-field-children"
    >
      Interuptable Air Conditioning
      , 
      Residential Gas Heating
    </span>
  </div>
</div>
`;
